app_dart/test/service/scheduler_test.dart

```diff
@@ -2892,9 +2894,153 @@ targets:
+              documentId: argThat(anything, named: 'documentId'),
+              mask_fieldPaths: argThat(anything, named: 'mask_fieldPaths'),
+              $fields: argThat(anything, named: r'$fields'),
+            ),
+          ).thenAnswer((_) async {
+            return Document();
+          });
+          return resource;
+        });
+
+        scheduler = Scheduler(
+          cache: cache,
+          config: FakeConfig(
+            dbValue: db,
+            githubService: mockGithubService,
+            githubClient: MockGitHub(),
+            firestoreService: mockFirestoreService,
+          ),
+          buildStatusProvider: (_, __) => buildStatusService,
+          datastoreProvider: (DatastoreDB db) => DatastoreService(db, 2),
+          githubChecksService: GithubChecksService(
+            config,
+            githubChecksUtil: mockGithubChecksUtil,
+          ),
+          getFilesChanged: getFilesChanged,
+          httpClientProvider: () => httpClient,
+          luciBuildService: fakeLuciBuildService,
+          fusionTester: fakeFusion,
+        )..firestoreService = mockFirestoreService;
+      });
+
+      tearDown(() {
+        printOnFailure('LOGGER BUFFER:\n${logs.join('\n')}');
+      });
+
+      test('does not run if running outside of Fusion (legacy releases)', () async {
+        fakeFusion.isFusion = (_, __) => false;
+        getFilesChanged.cannedFiles = ['packages/flutter/lib/material.dart'];
+        final pullRequest = generatePullRequest(authorLogin: allowListedUser);
+
+        await scheduler.triggerPresubmitTargets(pullRequest: pullRequest);
+
+        // Verify we don't try writing anything to Firestore (Fusion workflow).
+        verifyZeroInteractions(mockFirestoreService);
+        expect(
+          fakeLuciBuildService.scheduledTryBuilds.map((t) => t.value.name),
+          ['Linux A'],
+          reason: 'Should ignore the Fusion configuration and engine phases',
+        );
+      });
+
+      test('a non allow-listed user cannot use the optimization', () async {
+        fakeFusion.isFusion = (_, __) => true;
+        getFilesChanged.cannedFiles = ['packages/flutter/lib/material.dart'];
+        final pullRequest = generatePullRequest(authorLogin: 'joe-flutter');
+
+        await scheduler.triggerPresubmitTargets(pullRequest: pullRequest);
+        expect(
+          fakeLuciBuildService.scheduledTryBuilds.map((t) => t.value.name),
+          ['Linux engine_build'],
+          reason: 'Should still run engine phase',
+        );
+      });
+
+      test('an allow-listed user can use the optimization', () async {
+        fakeFusion.isFusion = (_, __) => true;
+        getFilesChanged.cannedFiles = ['packages/flutter/lib/material.dart'];
+        final pullRequest = generatePullRequest(authorLogin: allowListedUser);
+
+        await scheduler.triggerPresubmitTargets(pullRequest: pullRequest);
+        expect(
+          fakeLuciBuildService.scheduledTryBuilds.map((t) => t.value.name),
```

Comment:
can we verify the flutterPrebuiltEngineVersion was set to the base ref?

Reply:
Done.